.cicd/helpers/clang.make

```diff
@@ -3,9 +3,10 @@ set(CMAKE_CXX_COMPILER_WORKS 1)
 set(CMAKE_C_COMPILER /usr/local/bin/clang)
 set(CMAKE_CXX_COMPILER /usr/local/bin/clang++)
 set(CMAKE_CXX_STANDARD_INCLUDE_DIRECTORIES /usr/local/include/c++/v1 /usr/local/include /usr/include)
-set(CMAKE_C_FLAGS_INIT "-D_FORTIFY_SOURCE=2 -fstack-protector-strong -fpie")
-set(CMAKE_CXX_FLAGS_INIT "-nostdinc++ -D_FORTIFY_SOURCE=2 -fstack-protector-strong -fpie")
-set(CMAKE_EXE_LINKER_FLAGS_INIT "-stdlib=libc++ -nostdlib++ -pie")
+set(CMAKE_POSITION_INDEPENDENT_CODE ON)
```

Comment:
reason for this change?

Reply:
The `-fpie` would cause compiler warning in certain cases. See https://buildkite.com/EOSIO/eosio/builds/31652#96473ad1-aeb6-4b57-9827-0663654c7a86 which doesn’t have the change. Using `CMAKE_POSITION_INDEPENDENT_CODE` can make sure the cmake to use `-fpic` or `-fpie` correctly in all cases regardless building library or executable.

Reply:
```
[2021-12-14T18:25:31Z] /usr/bin/ld: CMakeFiles/test_abieos_reflect.dir/src/reflect_test.cpp.o: relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a PIE object; recompile with -fPIE
[2021-12-14T18:25:31Z] clang-10: error: linker command failed with exit code 1 (use -v to see invocation)
```
That looks like a linker error though?

Reply:
https://cmake.org/cmake/help/latest/policy/CMP0077.html
CMake version 3.22.1  warns when the policy is not set.

Reply:
The problem was cmake couldn't generate correct compile options. Here are the build.ninja snippets in both cases: 
- with `CMAKE_POSITION_INDEPENDENT_CODE=ON`
```
build libraries/abieos/CMakeFiles/test_abieos_reflect.dir/src/reflect_test.cpp.o: CXX_COMPILER__test_abieos_reflect ../libraries/abieos/src/reflect_test.cpp || cmake_object_order_depends_target_test_abieos_reflect
  DEP_FILE = libraries/abieos/CMakeFiles/test_abieos_reflect.dir/src/reflect_test.cpp.o.d
  FLAGS = -Werror -Wall -Wsign-compare -Wrange-loop-analysis -Wno-unused-command-line-argument -Wall -Wno-invalid-partial-specialization -O3 -DNDEBUG -fPIE    -fdiagnostics-color=always -std=gnu++17
  INCLUDES = -I../libraries/abieos/include -isystem /usr/local/include/c++/v1 -isystem /usr/local/include -isystem /usr/include
  OBJECT_DIR = libraries/abieos/CMakeFiles/test_abieos_reflect.dir
  OBJECT_FILE_DIR = libraries/abieos/CMakeFiles/test_abieos_reflect.dir/src
```
- With original CMAKE_C_FLAGS_INIT, CMAKE_CXX_FLAGS_INIT and CMAKE_EXE_LINKER_FLAGS_INIT
```
build libraries/abieos/CMakeFiles/test_abieos_reflect.dir/src/reflect_test.cpp.o: CXX_COMPILER__test_abieos_reflect ../libraries/abieos/src/reflect_test.cpp || cmake_object_order_depends_target_test_abieos_reflect
  DEP_FILE = libraries/abieos/CMakeFiles/test_abieos_reflect.dir/src/reflect_test.cpp.o.d
  FLAGS = -Werror -Wall -Wsign-compare -Wrange-loop-analysis -Wno-unused-command-line-argument -Wall -Wno-invalid-partial-specialization -O3 -DNDEBUG    -fdiagnostics-color=always -std=gnu++17
  INCLUDES = -I../libraries/abieos/include -isystem /usr/local/include/c++/v1 -isystem /usr/local/include -isystem /usr/include
  OBJECT_DIR = libraries/abieos/CMakeFiles/test_abieos_reflect.dir
  OBJECT_FILE_DIR = libraries/abieos/CMakeFiles/test_abieos_reflect.dir/src
  TARGET_COMPILE_PDB = libraries/abieos/CMakeFiles/test_abieos_reflect.dir/
  TARGET_PDB = libraries/abieos/test_abieos_reflect.pdb
```